### PR TITLE
add RangeRow component

### PR DESCRIPTION
**Pull request by jackwherry**
### Requirements:
- [x] The row’s height should automatically fit its container like DataPack does.
- [x] The RangeBar should have a width that can be passed as a prop to the generic row component. Also, its height should either be a fraction of the row’s height (preferred) or have a fixed vertical padding.
  - I can't seem to pass a percentage to the `h` prop from RangeBar—any idea why that might be? Right now I have it hardcoded to 1em instead.  
- [x] Aside from the props mentioned in other bullet points, the row component should only take in a label/title, a signal name, an optional precision (default is 3 decimal places), and an optional background color. The value, units, min, and max can be determined using the signal name. Don’t worry about the outliers like fan speed, pack power out, and pack temperature in BatteryPack.js. These won’t be issues anymore.
- [x] The text size/line height should be adjustable via an optional prop passed to the generic row component, or it could be automatically resized according to the nice-to-have below.
  - Do you think a small/medium/large would be appropriate here? I could then use a conditional to map the different size options into text sizes and row heights. How were you thinking this would work?
    - This is how I ended up doing it as of 4/13—it's definitely serviceable and works with the Chakra UI text size scheme that already exists

### Nice-to-haves:
- [x] If the text size could be adjusted to automatically fit the height of the row component, that would be very nice, but don’t let it block the task.

## Screenshots
4/14 vertically centered:
image

4/14:
image

4/12:
image

### CODE REVIEW

File: Frontend/src/Components/Shared/RangeRow.js [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+        rangeBarHeight = '1em';
+    } else if (props.size === 'md') {
+        lineHeight = '1.6em';
+        textSize = 'md';
+        rangeBarHeight = '1.4em';
+    } else if (props.size === 'lg') {
+        lineHeight = '2em';
+        textSize = 'lg';
+        rangeBarHeight = '1.8em';
+    }
```
Comment (jrvollmer) [inline Frontend/src/Components/Shared/RangeRow.js]: Let's add a default case for if `props.size` isn't one of the proper options (the default size should be 'sm')

File: Frontend/src/Components/Shared/RangeRow.js [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+
+export default function RangeRow(props) {
+    const { colorMode } = useColorMode();
+    const borderColor = getColor("border", colorMode);
+
+    const bg = (props.dataValue < props.dataConstant.MIN || props.dataValue > props.dataConstant.MAX) ?
+        getColor("errorBg", colorMode) : props.bg;
+
+    // set the decimal point to 3 by default
+    const decPoint = (props.DecimalPoint !== undefined) ? props.DecimalPoint : 3;
```
Comment (jrvollmer) [inline Frontend/src/Components/Shared/RangeRow.js]:
```suggestion
    const decPoint = props.DecimalPoint ?? 3;
```
Mostly for consistency, let's use this shorthand expression for undefined variables

File: Frontend/src/Components/Shared/RangeRow.js [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+    
+    let lineHeight;
+    let textSize;
+    let rangeBarHeight;
+    if (props.size === 'sm' || props.size === undefined) {
```
Comment (jrvollmer) [inline Frontend/src/Components/Shared/RangeRow.js]:
```suggestion
    if ((props.size ?? 'sm') === 'sm') {
```
Again, mostly for consistency, let's use the shorthand for handling undefined variables

**Review by VIkranthMandadi — APPROVED** [review]
what a beautiful component thank you for this jack

File: Frontend/src/Components/Shared/RangeRow.js [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+                <Spacer />
+                <Text fontSize={textSize} paddingRight='0.5em'>
+                    {props.dataValue.toFixed(decPoint)} {props.dataConstant.UNIT}
+                </Text>
+
+                {/* Hold the RangeBar within a vertical Flex to vertically center it */}
+                <Flex direction='column'>
+                    <Spacer />
+                    <RangeBar 
+                        w={props.w ?? '3em'}
```
Comment (jrvollmer) [inline Frontend/src/Components/Shared/RangeRow.js]:
```suggestion
                        w={props.rangeBarW ?? '3em'} 
```
Let's change the `w` prop to `rangeBarW`

**Review by jrvollmer — CHANGES_REQUESTED** [review]
You'll need to rebase off of main (and run `npm run update-data` from the root directory to make sure that the data format doesn't change after running that).

File: Frontend/src/Components/Shared/RangeRow.js (hunk 1 above)
Comment (jackwherry) [reply Frontend/src/Components/Shared/RangeRow.js]: Line 18 should set `sm` as the default already—do you think I should be doing it another way?

Comment (jackwherry) [reply Frontend/src/Components/Shared/RangeRow.js]: Never mind, just saw your later comment, I'll switch to using that syntax

Comment (jrvollmer) [reply Frontend/src/Components/Shared/RangeRow.js]: Sorry, I was proposing that there be a default case if a value was passed for `size`, but wasn't a valid value. However, that case should result in weird behavior, so you can ignore this thread.

**Comment by jrvollmer** [comment]
Also, @jackwherry, when you address the changes above (and double-check that this branch is off the most recent state of `main`), please squash your commits into one for this whole PR (named something like "Added RangeRow component"). @VIkranthMandadi @sarthyparty can you guys rebase off of this branch once that's done? Or, instead of rebasing, it's honestly probably going to be easier to just create a new branch off of that and then do the task again.

**Review by jrvollmer — APPROVED** [review]

**Pull request closed without merging**